components/automate-ui/src/app/page-components/projects-filter-dropdown/projects-filter-dropdown.component.ts

```diff
@@ -26,21 +26,31 @@ export class ProjectsFilterDropdownComponent implements OnChanges {
   @Input() filterVisible = false;
 
   @Output() onSelection = new EventEmitter<ProjectsFilterOption[]>();
+  @Output() onOptionChange = new EventEmitter<ProjectsFilterOption[]>();
 
   editableOptions: ProjectsFilterOption[] = [];
+  filteredOptions: ProjectsFilterOption[] = [];
+
+  filterValue = '';
 
   optionsEdited = false;
 
   dropdownActive = false;
 
+  handleFilterKeyUp() {
+    this.filteredOptions = this.editableOptions.filter(option =>
+      option.label.toLowerCase().indexOf(this.filterValue.toLowerCase()) > -1
```

Comment:
might help readability to move this chain of commands into a helper function with a name that explains what's happening to the input

Reply:
thanks for the suggestion, I added helper function to take care of the filtering now